xml_converter/integration_tests/src/trail_utils.py

```diff
@@ -0,0 +1,46 @@
+from .proto_utils import compare_binary_file
+from typing import List
+import difflib
+import os
+import struct
+
+
+def compare_trails(
+    expected_trail_path: str,
+    actual_trail_path: str
+) -> List[str]:
+    files_are_equal = compare_binary_file(expected_trail_path, actual_trail_path)
+
+    if files_are_equal:
+        return []
+
+    expected_textproto = get_texttrail(expected_trail_path)
+    actual_textproto = get_texttrail(actual_trail_path)
+
+    diff = list(difflib.unified_diff(actual_textproto.split("\n"), expected_textproto.split("\n"), fromfile=actual_trail_path, tofile=expected_trail_path, lineterm=""))
+
+    if len(diff) == 0:
+        diff = ["Something went wrong diffing {} and {}.".format(expected_trail_path, actual_trail_path)]
+
+    return list(diff)
+
+
+def get_texttrail(trail_path: str) -> str:
```

Comment:
This name seems odd. Could be `get_text_trail`, or `get_text_from_trail` and `get_trail_text`